l10n_it_ricevute_bancarie/account/account.py

```diff
@@ -85,3 +86,29 @@ class account_invoice(osv.osv):
     _columns = {
         'unsolved_move_line_ids': fields.many2many('account.move.line', 'invoice_unsolved_line_rel', 'invoice_id', 'line_id', 'Unsolved journal items'),
         }
+
+    def action_cancel(self, cr, uid, ids, context=None):
+        if ids:
+            invoice = self.browse(cr, uid, ids)[0]
+            # we get move_lines with date_maturity and check if they are
+            # present in some riba_distinta_line
+            move_line_model = self.pool['account.move.line']
+            rdml_model = self.pool['riba.distinta.move.line']
+            move_line_ids = move_line_model.search(
+                cr, uid, [('move_id', '=', invoice.move_id.id),
+                          ('date_maturity', '!=', False)])
```

Comment:
when you validate an invoice without payment terms, (receivable) move lines takes the invoice date for date_maturity